app/components/channel_icon/index.tsx

```diff
@@ -0,0 +1,186 @@
+    const styles = getStyleSheet(theme);
+
+    let activeIcon;
+    let unreadIcon;
+    let activeGroupBox;
+    let unreadGroupBox;
+    let activeGroup;
+    let unreadGroup;
+
+    if (isUnread) {
+        unreadIcon = styles.iconUnread;
+        unreadGroupBox = styles.groupBoxUnread;
+        unreadGroup = styles.groupUnread;
+    }
+
+    if (isActive) {
+        activeIcon = styles.iconActive;
+        activeGroupBox = styles.groupBoxActive;
+        activeGroup = styles.groupActive;
+    }
+
+    if (isInfo) {
+        activeIcon = styles.iconInfo;
+        activeGroupBox = styles.groupBoxInfo;
+        activeGroup = styles.groupInfo;
+    }
+
+    let icon;
+    if (isArchived) {
+        icon = (
+            <CompassIcon
+                name='archive-outline'
+                style={[styles.icon, unreadIcon, activeIcon, {fontSize: size, left: 1}]}
+                testID={`${testID}.archive`}
+            />
+        );
+    } else if (hasDraft) {
+        icon = (
+            <CompassIcon
+                name='pencil-outline'
+                style={[styles.icon, unreadIcon, activeIcon, {fontSize: size, left: 2}]}
+                testID={`${testID}.draft`}
+            />
+        );
+    } else if (shared) {
+        const iconName = type === General.PRIVATE_CHANNEL ? 'circle-multiple-outline-lock' : 'circle-multiple-outline';
+        const sharedTestID = type === General.PRIVATE_CHANNEL ? 'channel_icon.shared_private' : 'channel_icon.shared_open';
+        icon = (
+            <CompassIcon
+                name={iconName}
+                style={[styles.icon, unreadIcon, activeIcon, {fontSize: size, left: 0.5}]}
+                testID={sharedTestID}
+            />
+        );
+    } else if (type === General.OPEN_CHANNEL) {
+        icon = (
+            <CompassIcon
+                name='globe'
+                style={[styles.icon, unreadIcon, activeIcon, {fontSize: size, left: 1}]}
+                testID={`${testID}.public`}
+            />
+        );
+    } else if (type === General.PRIVATE_CHANNEL) {
+        icon = (
+            <CompassIcon
+                name='lock-outline'
+                style={[styles.icon, unreadIcon, activeIcon, {fontSize: size, left: 0.5}]}
+                testID={`${testID}.private`}
+            />
+        );
+    } else if (type === General.GM_CHANNEL) {
+        const fontSize = size - 12;
+        const boxSize = size - 4;
+        icon = (
+            <View
+                style={[styles.groupBox, unreadGroupBox, activeGroupBox, {width: boxSize, height: boxSize}]}
+            >
+                <Text
+                    style={[styles.group, unreadGroup, activeGroup, {fontSize}]}
+                    testID={`${testID}.gm_member_count`}
+                >
+                    {membersCount}
+                </Text>
+            </View>
+        );
+    } else if (type === General.DM_CHANNEL) {
+        //todo: Implement ProfilePicture component
```

Comment:
keep track of this so you include it as soon as the ProfilePicture component is added

Reply:
Yes I will.